clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -216,8 +216,10 @@
     private static final String MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION_DOC = "The maximum number of unacknowledged requests the client will send on a single connection before blocking."
                                                                             + " Note that if this config is set to be greater than 1 and <code>enable.idempotence</code> is set to false, there is a risk of"
                                                                             + " message re-ordering after a failed send due to retries (i.e., if retries are enabled)."
-                                                                            + " Additionally, enabling idempotence requires this config value to be less than or equal to " + MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION_FOR_IDEMPOTENCE + "."
-                                                                            + " If conflicting configurations are set and idempotence is not explicitly enabled, idempotence is disabled.";
+                                                                            + " Additionally, enabling idempotence requires the value of this configuration to be less than or equal to " + MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION_FOR_IDEMPOTENCE + "."
+                                                                            + " If conflicting configurations are set and idempotence is not explicitly enabled, idempotence is disabled. "
+                                                                            + " Record ordering is preserved when <code>enable.idempotence</code> is set to true for idempotent "
+                                                                            + " producer (or transactional producer), even when max in-flight requests are greater than 1 (supported up to 5).";
```

Comment:
Actually, I think the original version is better. Could you revert this change, and only fix the typo in stream config?

Reply:
Hello and thanks for reviewing!  The callout about record ordering was originally requested within Confluent; the same is mentioned at this page: https://developer.confluent.io/tutorials/message-ordering/kafka.html.  We think this would also benefit AK docs.

Reply:
The last sentence is redundant though, isn't it? The docs already state:

> Note that if this config is set to be greater than 1 and <code>enable.idempotence</code> is set to false, there is a risk of message re-ordering after a failed send due to retries (i.e., if retries are enabled).

It's fine if we want to clarify that re-ordering is not a risk when `enable.idempotence` is set to true, but we should also try to keep the docs here concise.

What about adding a brief followup in that sentence instead?

> Note that if this config is set to be greater than 1 and <code>enable.idempotence</code> is set to false, there is a risk of message re-ordering after a failed send due to retries (i.e., if retries are enabled); ordering will be preserved if retries are disabled and/or <code>enable.idempotence</code> is true.

Reply:
@C0urante i have updated the text with your feedback. thanks.